app/custom_auth.py

```diff
@@ -0,0 +1,57 @@
+from django.contrib import auth
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.exceptions import PermissionDenied
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+class CustomSSOMiddleware:
```

Comment:
Why aren't you subclassing RemoteUserMiddleware here as in [Authenication using Remote user](https://docs.djangoproject.com/en/2.2/howto/auth-remote-user/)? Is the remote user middleware being used currently?

Reply:
We are not creating a new backend and using the default backend that comes with Django.

Reply:
To Arman's point, the [RemoteUserMiddleware or PersistenRemoteUserMiddleware](https://github.com/django/django/blob/402e6d292fd8aa5dfd49166e80098122950053f7/django/contrib/auth/middleware.py#L27-L123) looks essentially like what we implemented here.